mungegithub/mungers/lgtm_handler.go

```diff
@@ -0,0 +1,166 @@
+package mungers
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"k8s.io/contrib/mungegithub/features"
+	"k8s.io/contrib/mungegithub/github"
+	"k8s.io/contrib/mungegithub/mungers/mungerutil"
+
+	"github.com/golang/glog"
+	githubapi "github.com/google/go-github/github"
+	"github.com/spf13/cobra"
+)
+
+const (
+	lgtmRemovedBody = "PR changed after LGTM, removing LGTM."
+)
+
+// LGTMHandler will
+// - apply the LGTM label if reviewer has said so, or
+// - remove the LGTM label from an PR which has been updated since the reviewer added LGTM
+type LGTMHandler struct{}
+
+func init() {
+	l := LGTMHandler{}
+	RegisterMungerOrDie(l)
+	RegisterStaleComments(l)
+}
+
+// Name is the name usable in --pr-mungers
+func (LGTMHandler) Name() string { return "lgtm-after-commit" }
+
+// RequiredFeatures is a slice of 'features' that must be provided
+func (LGTMHandler) RequiredFeatures() []string { return []string{} }
+
+// Initialize will initialize the munger
+func (LGTMHandler) Initialize(config *github.Config, features *features.Features) error {
+	return nil
+}
+
+// EachLoop is called at the start of every munge loop
+func (LGTMHandler) EachLoop() error { return nil }
+
+// AddFlags will add any request flags to the cobra `cmd`
+func (LGTMHandler) AddFlags(cmd *cobra.Command, config *github.Config) {}
+
+// Munge is the workhorse the will actually make updates to the PR
+func (h LGTMHandler) Munge(obj *github.MungeObject) {
+	if !obj.IsPR() {
+		return
+	}
+
+	lastModified := obj.LastModifiedTime()
+	if !obj.HasLabel(lgtmLabel) {
+		allAssignees := append(obj.Issue.Assignees, obj.Issue.Assignee)
+		reviewers := mungerutil.FilterValid(allAssignees)
+		if len(reviewers) == 0 {
+			return
+		}
+		comments, err := obj.ListComments()
+		if err != nil {
+			glog.Errorf("unexpected error getting comments: %v", err)
+		}
+		users := findLGTMCommenters(comments, lastModified)
+		for _, user := range users {
+			// TODO: support more complex policies for multiple reviewers.
+			// See https://github.com/kubernetes/contrib/issues/1389#issuecomment-235161164
+			if isReviewer(user, reviewers) {
+				obj.AddLabel(lgtmLabel)
+				return
+			}
+		}
+		if len(users) != 0 {
+			// if somebody that is NOT the reviewer says "LGTM", it would be nice to mention the reviewer, rather than just ignore.
+			obj.WriteComment(fmt.Sprintf("LGTM from users (%v). Can reviewers (%v) review this?", users, reviewers))
+		}
+		return
+	}
+
+	lgtmTime := obj.LabelTime(lgtmLabel)
+
+	if lastModified == nil || lgtmTime == nil {
+		glog.Errorf("PR %d unable to determine lastModified or lgtmTime", *obj.Issue.Number)
+		return
+	}
+
+	if lastModified.After(*lgtmTime) {
+		glog.Infof("PR: %d lgtm:%s  lastModified:%s", *obj.Issue.Number, lgtmTime.String(), lastModified.String())
+		if err := obj.WriteComment(lgtmRemovedBody); err != nil {
+			return
+		}
+		obj.RemoveLabel(lgtmLabel)
+	}
+}
+
+func findLGTMCommenters(comments []*githubapi.IssueComment, lastModified *time.Time) (res []*githubapi.User) {
```

Comment:
> Are comments sorted?

https://developer.github.com/v3/issues/comments/#list-comments-on-an-issue
"Issue Comments are ordered by ascending ID."

> It'd be nice to leave the loop early if possible

If we sort it ourselves, looping is O(N) and sorting is O(N*lgN). Instead, why don't we have metrics system?